api/src/main/java/org/apache/iceberg/catalog/TableIdentifier.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.catalog;
+
+/**
+ * Identifies a table in iceberg catalog
+ */
+public class TableIdentifier {
+  private final Namespace namespace;
+  private final String name;
+
+  public TableIdentifier(Namespace namespace, String name) {
```

Comment:
Should we also have a constructor without a namespace? That would call this one with a static empty namespace (0 levels).

Reply:
I will change this to Optional<Namespace> and define 2 constructors.

Reply:
added factory in namespace and helper methods so caller can easily check if namespace is empty or not.